ci/Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,445 @@
+  return "${BRANCH_NAME}-" + getCurrentVersion()
+}
+
+String getCurrentVersion() {
+  return readMavenPom().getVersion();
+}
+
+String getReleaseVersion() {
+  container('maven') {
+    String nuxeoVersion = getCurrentVersion()
+    String noSnapshot = nuxeoVersion.replace('-SNAPSHOT', '')
+    String version = noSnapshot + '.1' // first version ever
+    // find the latest tag if any
+    sh """
+      git fetch origin 'refs/tags/v${noSnapshot}*:refs/tags/v${noSnapshot}*'
+    """
+    def tag = sh(returnStdout: true, script: "git tag --sort=taggerdate --list 'v${noSnapshot}*' | tail -1 | tr -d '\n'")
+    if (tag) {
+      version = sh(returnStdout: true, script: "semver bump patch ${tag} | tr -d '\n'")
+    }
+    return version
+  }
+}
+
+void setGitHubBuildStatus(String context, String message, String state) {
+  if (env.DRY_RUN != "true") {
+    step([
+      $class            : 'GitHubCommitStatusSetter',
+      reposSource       : [$class: 'ManuallyEnteredRepositorySource', url: repositoryUrl],
+      contextSource     : [$class: 'ManuallyEnteredCommitContextSource', context: context],
+      statusResultSource: [$class: 'ConditionalStatusResultSource', results: [[$class: 'AnyBuildResult', message: message, state: state]]],
+    ])
+  }
+}
+
+String getVersion() {
+  return isPullRequest() ? getPullRequestVersion() : getReleaseVersion()
+}
+
+void helmfileSync(namespace, environment) {
+  withEnv(["NAMESPACE=${namespace}"]) {
+    sh """
+      ${HELMFILE_COMMAND} deps
+      ${HELMFILE_COMMAND} --environment ${environment} sync
+    """
+  }
+}
+
+void helmfileDestroy(namespace, environment) {
+  withEnv(["NAMESPACE=${namespace}"]) {
+    sh """
+      ${HELMFILE_COMMAND} --environment ${environment} destroy
+    """
+  }
+}
+
+def buildUnitTestStage(env) {
+  def isDev = env == 'dev'
+  def testNamespace = "${TEST_NAMESPACE_PREFIX}-${env}"
+  // Helmfile environment
+  def environment = "${env}UnitTests"
+  def kafkaHost = "${TEST_KAFKA_K8S_OBJECT}.${testNamespace}.${TEST_SERVICE_DOMAIN_SUFFIX}:${TEST_KAFKA_PORT}"
+  def containerName = isDev ? 'maven' : "maven-${env}"
+  return {
+    stage("Run ${env} unit tests") {
+      container("${containerName}") {
+        // TODO NXP-29512: on a PR, make the build continue even if there is a test error
+        // on other environments than the dev one
+        // to remove when all test environments will be mandatory
+        catchError(buildResult: isPullRequest() && "dev" != env ? 'SUCCESS' : 'FAILURE', stageResult: 'FAILURE', catchInterruptions: false) {
+          script {
+            setGitHubBuildStatus("utests/${env}", "Unit tests - ${env} environment", 'PENDING')
+            sh "kubectl create namespace ${testNamespace}"
+            try {
+              echo """
+              ----------------------------------------
+              Run ${env} unit tests
+              ----------------------------------------"""
+
+              echo "${env} unit tests: run Maven"
```

Comment:
this should in fact be displayed just before running the Maven command, above `sh "${mvnCommand}"`